### ISSUE TITLE
is "don't use it" warning still accurate?

### ISSUE THREAD

**Issue by mogest** [gabrielelana/paco#4]
After searching far and wide for a good Elixir parser, paco is it.  I've been using it for my Advent of Code solutions, and it's working very, very nicely.

However it looks like paco hasn't been touched for 15+ months, and it still has a warning in the readme about how it shouldn't be used.

I'm interested in your future plans for paco.  What's keeping you from removing that warning?  If work needs to be done, are you interested in pull requests?

**Comment by gabrielelana** [comment]
> After searching far and wide for a good Elixir parser, paco is it. I've been using it for my Advent of Code solutions, and it's working very, very nicely.

Thanks! Happy to hear that

> However it looks like paco hasn't been touched for 15+ months, and it still has a warning in the readme about how it shouldn't be used.

It works but I'm not happy with it and I hate to publish half baked work

> I'm interested in your future plans for paco. What's keeping you from removing that warning?

Two biggest problems right now are (see last slide of http://www.slideshare.net/gabriele.lana/parse-everything-with-elixir):
- Support for left recursive grammars
- Performance

Both can be solved with some sort of memoization, I know how to do it, I have a working prototype but it's kind of a major rewrite so I need to find the time to do it

Knowing that this could be useful I might find the time ;-)

> If work needs to be done, are you interested in pull requests?

After this major thing there will be work to do:
- more examples
- more combinators
- more parsers to reuse
- documentation

If you are still available by then I will gladly accept help

**Comment by mogest** [comment]
Personally, I don't think it's half-baked.  It might not have all the functionality you were looking for, but it's going to work perfectly for most use cases!

Regarding the rewrite, do you think you'll change the function signatures as part of that?  I ask because if you intend to keep them (mostly) the same, I could start writing up some documentation now.  The only challenge I had in using it was working out what everything did; I found that slidedeck above and worked it out from them plus looking at the source.

**Comment by gabrielelana** [comment]
>  do you think you'll change the function signatures as part of that?

Yes, they will stay mostly the same, in particular the combinators (the parsers in the `parser.ex` file) will not change, at least in what they do

>  I could start writing up some documentation now

That would be wonderful 😄 

> The only challenge I had in using it was working out what everything did

Yep, the best sources of knowledge right now are the [tests](https://github.com/gabrielelana/paco/tree/master/test/parser) and the [examples](https://github.com/gabrielelana/paco/tree/master/examples)

If you have any questions please ask, I will resume the project in the following days

Thanks again for your interest

**Comment by tdejager** [comment]
Hi!

Already, thanks for the awesome work 👍 I too would like to know if a version could be published on hex sometime. I think that this parser library, in comparison to other parser combinator libraries like 'combine' for example, does have an advantage and I believe it would benefit the elixir eco-system!

:)

**Comment by gabrielelana** [comment]
@tdejager thanks, I'll try to publish it in the following days, stay tuned 😎
